### ISSUE TITLE
Updating from torch 2.8 to 2.9 breaks loss/gradients

### ISSUE THREAD

**Issue by Kefir101** [pytorch/pytorch#174441]
### 🐛 Describe the bug

After updating from 2.8.0 to 2.9.0, mixed precision seems to overflow. On pre-trained models like ResNet18 with Adam and with a 0.001 learning rate, even though training loss goes towards convergence, validation loss NaNs out at around one in 10 epochs. Very rarely, the training loss NaNs out too. 

If we switch to non-mixed precision, there are no more NaN's, but training takes 5-6x as long.

```py

   # main.py
   trainer = L.Trainer(
        precision="16-mixed",
        logger=logger,
        max_epochs=maxepochs,
        min_epochs=minepochs,
        callbacks=callbacks,
        enable_checkpointing=not disable_model_saves,
    )
  
    # model.py
    def step(self, batch, batch_idx):
        inputs, target, weight = batch
        output = self(inputs)
        loss = nn.functional.binary_cross_entropy_with_logits(
            output.squeeze(), target, weight=weight
        )
        preds = torch.sigmoid(output.squeeze())
        return loss, preds, target

    def training_step(self, batch, batch_idx):
        train_loss, preds, target = self.step(batch, batch_idx)
        return self.get_data_dict(
            "train", self.train_metrics, train_loss, preds, target
        )

    def validation_step(self, batch, batch_idx):
        val_loss, preds, target = self.step(batch, batch_idx)
        return self.get_data_dict("val", self.val_metrics, val_loss, preds, target)
```

### Versions

Collecting environment information...
PyTorch version: 2.8.0+cu128
Is debug build: False
CUDA used to build PyTorch: 12.8
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.3 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.39

Python version: 3.12.3 (main, Jan  8 2026, 11:30:50) [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-6.14.0-1018-aws-x86_64-with-glibc2.39
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: Tesla T4
Nvidia driver version: 580.65.06
cuDNN version: Could not collect
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Architecture:                            x86_64
CPU op-mode(s):                          32-bit, 64-bit
Address sizes:                           46 bits physical, 48 bits virtual
Byte Order:                              Little Endian
CPU(s):                                  32
On-line CPU(s) list:                     0-31
Vendor ID:                               GenuineIntel
Model name:                              Intel(R) Xeon(R) Platinum 8259CL CPU @ 2.50GHz
CPU family:                              6
Model:                                   85
Thread(s) per core:                      2
Core(s) per socket:                      16
Socket(s):                               1
Stepping:                                7
BogoMIPS:                                4999.99
Flags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch cpuid_fault pti fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid mpx avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves ida arat pku ospke avx512_vnni
Hypervisor vendor:                       KVM
Virtualization type:                     full
L1d cache:                               512 KiB (16 instances)
L1i cache:                               512 KiB (16 instances)
L2 cache:                                16 MiB (16 instances)
L3 cache:                                35.8 MiB (1 instance)
NUMA node(s):                            1
NUMA node0 CPU(s):                       0-31
Vulnerability Gather data sampling:      Unknown: Dependent on hypervisor status
Vulnerability Ghostwrite:                Not affected
Vulnerability Indirect target selection: Mitigation; Aligned branch/return thunks
Vulnerability Itlb multihit:             KVM: Mitigation: VMX unsupported
Vulnerability L1tf:                      Mitigation; PTE Inversion
Vulnerability Mds:                       Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown
Vulnerability Meltdown:                  Mitigation; PTI
Vulnerability Mmio stale data:           Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown
Vulnerability Reg file data sampling:    Not affected
Vulnerability Retbleed:                  Vulnerable
Vulnerability Spec rstack overflow:      Not affected
Vulnerability Spec store bypass:         Vulnerable
Vulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:                Mitigation; Retpolines; STIBP disabled; RSB filling; PBRSB-eIBRS Not affected; BHI Retpoline
Vulnerability Srbds:                     Not affected
Vulnerability Tsa:                       Not affected
Vulnerability Tsx async abort:           Not affected
Vulnerability Vmscape:                   Not affected

Versions of relevant libraries:
[pip3] Could not collect
[conda] Could not collect

cc @mcarilli @ptrblck @leslie-fang-intel @jgong5

**Comment by jbschlosser** [comment]
Hi, I suggest opening a conversion on the [PyTorch forums](https://discuss.pytorch.org/), as you're more likely to receive assistance there. Closing for now but feel free to reopen with more details if discussion there indicates a problem within PyTorch itself.
